### PR TITLE
User OAUTH code

### PR DESCRIPTION
**Pull request by GlazerMann**
@ms609  needs to upgrade the exising OAuth tokens so that they can also get user tokens at https://meta.wikimedia.org/wiki/Special:OAuthConsumerRegistration.  Once they are valid for users.  Then we get to debug on the development branch.  You will also have to specify a return URL: I chose authenticate.php

Unknown:  Do we need a new edit token for editing of a file.  ie. do we need to grab a new edit token for each and every file in a category?

Once we can do that, we should tag our edits (this is can be setup to automatically do this based upon our oauth keys) https://en.wikipedia.org/wiki/Special:Tags

### CODE REVIEW

**Comment by pppery** [comment]
If you're using OAuth, wouldn't it be better to go all-out and actually make the edit using the activating user's account instead of the bot's?

**Comment by GlazerMann** [comment]
That’s exactly what this pull does @pppery

**Comment by pppery** [comment]
Sorry, I'm not very good at reading complicated PHP code.

**Comment by GlazerMann** [comment]
@ms609 people are grumpy about the anonymous nature and I am using it as a learning opportunity.  No rush at all.  Some more ideas hit me recently on this and I have some changes to make.

**Comment by GlazerMann** [comment]
@ms609 there is a movement to PERMANENTLY BLOCK us unless be stop all anonymous editing.  We need to get this working on the dev bot.

**Comment by GlazerMann** [comment]
https://en.wikipedia.org/w/index.php?title=Wikipedia:Bot_policy&diff=next&oldid=204147936 

The bots current behavior violates Wikipedia policy that changed a decade ago

**Comment by GlazerMann** [comment]
TODO: Dump all the header stuff and force people to click a link

**Comment by ms609** [comment]
I'm not sure how to upgrade OAuth tokens; the 'manage tokens' page doesn't include such an option.  Could you provide a link?

**Comment by GlazerMann** [comment]
I think you have to go to https://meta.wikimedia.org/wiki/Special:OAuthConsumerRegistration as the citation bot account.  Not 100% sure

File: category.php [hunk 1]
```diff
@@ -47,7 +47,7 @@
 <?php
 }
 
-$user = isset($_REQUEST["user"]) ? $_REQUEST["user"] : NULL;
+$user = NULL; // Editing as user or script, no need to name.
 if (is_valid_user($user)) {
   echo " Activated by $user.\n\n";
   $edit_summary_end = " | [[User:$user|$user]]; [[Category:$category]].";
```
Comment (pppery) [inline category.php]: This is dead code, and, once the edits are done using OAUTH, there is no need to add "user-activated" to the edit summary (alternately, replace it with something like "using Citation bot")

Comment (GlazerMann) [reply category.php]: Original attempt was to minimize code changes

**Pull request closed without merging**